src/solvers/dgsem_t8code/containers_3d.jl

```diff
@@ -0,0 +1,232 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# Interpolate tree_node_coordinates to each quadrant at the specified nodes
+function calc_node_coordinates!(node_coordinates,
+                                mesh::T8codeMesh{3},
+                                nodes::AbstractVector)
+    # We use `StrideArray`s here since these buffers are used in performance-critical
+    # places and the additional information passed to the compiler makes them faster
+    # than native `Array`s.
+    tmp1 = StrideArray(undef, real(mesh),
+                       StaticInt(3), static_length(nodes), static_length(mesh.nodes),
+                       static_length(mesh.nodes))
+    matrix1 = StrideArray(undef, real(mesh),
+                          static_length(nodes), static_length(mesh.nodes))
+    matrix2 = similar(matrix1)
+    matrix3 = similar(matrix1)
+    baryweights_in = barycentric_weights(mesh.nodes)
+
+    num_local_trees = t8_forest_get_num_local_trees(mesh.forest)
+
+    current_index = 0
+    for itree in 0:(num_local_trees - 1)
+        tree_class = t8_forest_get_tree_class(mesh.forest, itree)
+        eclass_scheme = t8_forest_get_eclass_scheme(mesh.forest, tree_class)
+        num_elements_in_tree = t8_forest_get_tree_num_elements(mesh.forest, itree)
+
+        for ielement in 0:(num_elements_in_tree - 1)
+            element = t8_forest_get_element_in_tree(mesh.forest, itree, ielement)
+            element_level = t8_element_level(eclass_scheme, element)
+
+            element_length = t8_hex_len(element_level) / t8_hex_root_len
```

Comment:
Maybe add a comment here why one cannot simply use `t8_hex_len`?

Reply:
I added a comment to explain why the line is written as it is.